script/aria.js

```diff
@@ -7,746 +20,395 @@
-                    ">" +
-                    lItem.name +
-                    "</pref>";
-            }
-            if (lItem.prohibited) {
-                globalSPIndex += " (Except where prohibited)";
-            }
-            if (lItem.deprecated) {
-                globalSPIndex += " (Global use deprecated in ARIA 1.2)";
-            }
-            globalSPIndex += "</li>\n";
-        }
-        parentNode = document.querySelector("#global_states");
-        if (parentNode) {
-            node = parentNode.querySelector(".placeholder");
-            if (node) {
-                l = document.createElement("ul");
-                l.innerHTML = globalSPIndex;
-                parentNode.replaceChild(l, node);
-            }
-        }
-        // there is only one role that uses the global properties
-        parentNode = document.querySelector(
-            "#roletype td.role-properties span.placeholder"
-        );
-        if (parentNode) {
-            node = parentNode.parentNode;
-            if (
-                (parentNode.textContent || parentNode.innerText) ===
-                "Placeholder for global states and properties"
-            ) {
-                l = document.createElement("ul");
-                l.innerHTML = globalSPIndex;
-                node.replaceChild(l, parentNode);
-            }
-        }
+/**
+ *
+ * @param {HTMLElement} item - rdef element
+ */
+const rewriteRdef = function (item) {
+    // TODO: merge with generateHTMLStatesAndProperties() but that creates different HTML
+    const content = item.innerHTML;
+    let title = item.getAttribute("title") || content;
+    let type = "role";
+    const abstract = item.parentNode.querySelectorAll(".role-abstract"); //TODO: maybe #105
+    if (abstract.innerText === "True") {
+        type = "abstract role";
     }
+    const dRef = item.nextElementSibling;
+    dRef.id = "desc-" + title;
+    dRef.setAttribute("role", "definition");
+    item.outerHTML = `<h4 class="role-name" title="${title}" aria-describedby="desc-${title}"><code>${content}</code> <span class="type-indicator">${type}</span>`;
+};
 
-    // what about roles?
-    //
-    // we need to do a few things here:
-    //   1. expand the rdef elements.
-    //   2. accumulate the roles into a table for the indices
-    //   3. grab the parent role reference so we can build up the tree
-    //   4. grab any local states and properties so we can hand those down to the children
-    //
+/**
+ * Replaces sdef/pdef with desired HTML
+ * @param {Object} propList -
+ * @param {HTMLElement} item - sdef or pdef, from nodeList.forEach
+ */
+const generateHTMLStatesAndProperties = function (propList, item) {
+    const title = item.getAttribute("title") || item.innerHTML;
+    const itemEntry = propList[title];
+    const dRef = item.nextElementSibling;
+    dRef.id = "desc-" + title; // TODO: too much of a side-effect?
+    dRef.setAttribute("role", "definition"); // TODO: ditto?
```

Comment:
this kind of stuff is a little weird to be adding here, it's not generated, it should just be in the html, it seems.